soccer/gameplay/role_assignment.py

```diff
@@ -181,11 +181,13 @@ def assign_roles(robots, role_reqs):
     # To do our calculations though, we need to have a flat list of RoleRequirements to match to our list of robots
     # The flatten_tree() method does this and keeps track of how to rebuild the tree so we can do so at the end
     def flatten_tree(tree, path_prefix=[]):
+        valid_shell_ids = [bot.shell_id() for bot in robots]
         for key, subtree in tree.items():
             if isinstance(subtree, dict):
                 flatten_tree(subtree, path_prefix + [key])
             elif isinstance(subtree, RoleRequirements):
-                role_reqs_list.append(subtree)
+                if subtree.required_shell_id == None or subtree.required_shell_id in valid_shell_ids:
```

Comment:
@jgkamat Hmm... This is actually a little confusing. The way the system was originally set up, the c++ side used None if the goalie wasn't set and the python side used -1. I switched the python side to match the c++ side so that even if the external ref sends a weird value like -1 (which it can technically send) it won't break anything. As such, the value of required_shell_id is actually set to None while the value should technically exist. Does this mean I should use == since I am checking if the int actually is set to None and not that it doesn't exist?